tools/vanity/vanity/__main__.py

```diff
@@ -53,8 +53,9 @@ def main():
 		generator.match_all(matcher)
 	finally:
 		for outfile in printer.outfiles:
-			outfile.close()
+			if '<stdout>' != outfile.name:
+				outfile.close()
 
 
 if '__main__' == __name__:
-	main()
+	main(sys.argv[1:])
```

Comment:
hmm, why is this needed?

Reply:
because main() now takes a parameter so it can be called by tests (so argparse doesn't operate on global)